ols/src/query_helpers/docs_summarizer.py

```diff
@@ -80,20 +89,21 @@ def summarize(
                 logger.info(f"{conversation} Setting up index")
                 index = load_index_from_storage(
                     storage_context=storage_context,
-                    index_id=config.ols_config.reference_content.product_docs_index_id,
                     service_context=service_context,
+                    index_id=config.ols_config.reference_content.product_docs_index_id,
                     verbose=verbose,
                 )
                 logger.info(f"{conversation} Setting up query engine")
-                query_engine = index.as_query_engine(
-                    text_qa_template=summarization_template,
-                    verbose=verbose,
-                    streaming=False,
-                    similarity_top_k=1,
-                )
 
                 logger.info(f"{conversation} Submitting summarization query")
-                summary = query_engine.query(query)
+
+                chat_engine = index.as_chat_engine(
+                    service_context=service_context,
+                    chat_history=chat_history,
+                    chat_mode="context",
+                    system_prompt=summarization_template,
+                )
+                summary = chat_engine.chat(query)
```

Comment:
this is ok for now but sooner rather than later we're going to drop the use of llamaindex for our RAG-based query flows and instead construct the prompt(with rag content) ourselves, specifically as part of https://issues.redhat.com/browse/OLS-104

So eventually i expect this codepath to look more like the no-RAG flow below.